### PR TITLE
Disable send button on messages

### PR DESCRIPTION
**Pull request by rmccar**
### What is the context of this PR?
This PR makes changes to disable the button on messages if there is no text in the message box. This will stop any users submitting empty messages.

I have added a reply component to do this which is now used to send the messages

### How to review
Check messages work as it should disabling the submit button when there is no text in the message box

### CODE REVIEW

**Comment by rmccar** [comment]
[Send and reply to messages pattern: Disable send button until text entered](https://trello.com/c/lX8eAe90/1187-send-and-reply-to-messages-pattern-disable-send-button-until-text-entered)

File: src/components/autosuggest/_macro-options.md [hunk 1]
```diff
@@ -6,7 +6,7 @@
 | allowMultiple           | boolean           | false    | Allows the component to accept multiple selections                                                                                                 |
 | instructions            | string            | true     | Instructions on how to use the autosuggest that will be read out by screenreaders                                                                  |
 | ariaYouHaveSelected     | string            | true     | Aria message to tell the user that they have selected an answer                                                                                    |
-| ariaMinChars            | string            | true     | Aria message to tell the user how many charecters they need to enter before autosuggest will start                                                 |
+| ariaMinChars            | string            | true     | Aria message to tell the user how many character they need to enter before autosuggest will start                                                  |
```
Comment (jrbarnes9) [inline src/components/autosuggest/_macro-options.md:9]: Nearly ;)
```suggestion
| ariaMinChars            | string            | true     | Aria message to tell the user how many characters they need to enter before autosuggest will start                                                  |
```

File: src/components/input/_macro-options.md [hunk 2]
```diff
@@ -47,7 +47,7 @@
 | allowMultiple           | boolean           | false    | Allows the component to accept multiple selections                                                                                                 |
 | instructions            | string            | true     | Instructions on how to use the autosuggest that will be read out by screenreaders                                                                  |
 | ariaYouHaveSelected     | string            | true     | Aria message to tell the user that they have selected an answer                                                                                    |
-| ariaMinChars            | string            | true     | Aria message to tell the user how many charecters they need to enter before autosuggest will start                                                 |
+| ariaMinChars            | string            | true     | Aria message to tell the user how many character they need to enter before autosuggest will start                                                  |
```
Comment (jrbarnes9) [inline src/components/input/_macro-options.md:50]:
```suggestion
| ariaMinChars            | string            | true     | Aria message to tell the user how many characters they need to enter before autosuggest will start                                                  |
```

File: src/patterns/message/examples/conversation/index.njk [hunk 3]
```diff
@@ -100,29 +99,23 @@
...
+                            "text": "Reply"
+                        }
+                    },
+                    "button": {
+                        "type": 'button',
+                        "text": 'Send message',
+                        "classes": 'btn-group__btn'
+                    },
+                    "closeLinkText": 'Close conversation',
+                    "closeLinkUrl": '#'
```
Comment (jrbarnes9) [inline src/patterns/message/examples/conversation/index.njk:116]:
```suggestion
                    "closeLinkUrl": '#0'
```

File: src/tests/spec/reply/reply.spec.js [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    },
+  },
+  button: {
+    id: 'replyButton',
+    type: 'button',
+    text: 'Send message',
+    classes: 'btn-group__btn',
+  },
+  closeLinkText: 'Close conversation',
+  closeLinkUrl: '#',
```
Comment (jrbarnes9) [inline src/tests/spec/reply/reply.spec.js:20]:
```suggestion
  closeLinkUrl: '#0',
```

**Review by jrbarnes9 — COMMENTED** [review]
Can you also update the Discussions link and title to match the pattern name: "Send and reply to messages"

File: src/components/reply/reply-input.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    this.input = this.context.querySelector('.input');
+    this.button = this.context.querySelector('.btn');
+
+    this.enableDisableButton(this.input);
+
+    this.input.addEventListener('input', this.enableDisableButton.bind(this));
+  }
+
+  enableDisableButton() {
+    console.log(this.input.value);
```
Comment (boxadesign) [inline src/components/reply/reply-input.js:13]: remove log line

**Review by jrbarnes9 — APPROVED** [review]

**Review by boxadesign — APPROVED** [review]

**Pull request merged**